openid4vc-high-assurance-interoperability-profile-sd-jwt-vc-1_0.md

```diff
@@ -215,6 +223,57 @@ This is an example of a Wallet Instance Attestation:
     * In the `input-descriptors` object, `id`, `name`, `purpose`, `group`, `format`, and `constraints` properties MUST be supported. In the `constraints` object, `limit_disclosure`, and `fields` properties MUST be supported. In the `fields` object, `path` and `filter` properties MUST be supported. A `path` MUST contain exactly one entry with a static path to a certain claim. A `filter` MUST only contain `type` elements of value `string` and `const` elements.
     * In the `submission_requirements` object, `name`, `rule (`pick` only)`, `count`, `from` properties MUST be supported.
 
+# OpenID for Verifiable Presentations over W3C Digital Credentials API
+
+* MUST support Annex A in [@!OIDF.OID4VP] that defines how to use OID4VP over the W3C Digital Credentials API.
+  * Annex A.3.1. in [@!OIDF.OID4VP] MUST NOT be used. OID4VP requests over the W3C Digital Credentials API MUST be signed.
+* Wallet Invocation is done via the W3C Digital Credentials API or an equivalent platform API. Custom URL schemes MUST NOT be used.
+* Response Mode MUST be `w3c_dc_api.jwt`. Encryption of the response is mandatory.
+* MUST support Transaction Data defined in Sections 5.4 and 7.4 of [@!OIDF.OID4VP].
+* The DQCL query and response as defined in Section 6 of [@!OIDF.OID4VP] MUST be used. Presentation Exchange as defined in Sections 5.4 and 5.5 of [@!OIDF.OID4VP] MUST NOT be used. Below is the list of features in the DQCL query and response that MUST be supported:
+  * tbd
+
+## ISO mdoc specific requirements for OpenID for Verifiable Presentations over W3C Digital Credentials API
+
+  * The Credential format identifier MUST be `mso_mdoc`.
+  * mdoc Credential Format specific DCQL parameters as defined in Section 6.4.2. of [@!OIDF.OID4VP] MUST be used.
+  * Verifier MAY request more than one credential in the same request.
+  * DeviceResponse in the VP Token MUST contain only one mdoc. Therefore, when multiple mdocs are being returned, each mdoc should be returned in a separate DeviceResponse, each matching to a respective DCQL query.
+  * Response encryption MUST be done as defined in Section B.4.3.3.2 in [@!ISO.18013-7].
+
+### Session Transcript
+
+The SessionTranscript as defined in [@ISO.18013-5] shall be used with the following changes:
+
+* DeviceEngagementBytes MUST be null.
+* EReaderKeyBytes MUST ne null
+
+The Handover element is defined as following:
+
+```
+Handover = OID4VPDCAPIHandover
+OID4VPDCAPIHandover = [
+  "OID4VPDCAPIHandover"
+  clientId
```

Comment:
From a security perspective, what kind of attacks are mitigated by inclusion of the clientID in the sessiontranscript?
Inclusion of the clientId also has other issues, like which value is used if there are multiple signatures, or if the wallet ignores RP authentication.

Reply:
yes, we need clientId as audience restriction.

Reply:
The wallet must decide which audience a certain credential presentation is intended for, even in case of multiple signatures. It's a selection 1 out of n in case of multiple signatures.